menu.cpp

```diff
@@ -28,10 +28,21 @@ std::vector<menu_button> &menu::get_buttons() noexcept { return m_v_buttons; }
 
 menu_button &menu::get_button(int index)
 {
-
   assert(index >= 0 && index < static_cast<int>(m_v_buttons.size()));
   return m_v_buttons[static_cast<unsigned int>(index)];
+}
 
+menu_button &menu::get_button(const std::string& label)
+{
+  for (auto& mb : m_v_buttons)

```

Comment:
The indentation is a bit different than the usual, this is the most-user indentation structure:

```c++
for (..)
{
  if (...) 
  {

  }
}
```